geomesa-fs/geomesa-fs-tools/src/main/scala/org/locationtech/geomesa/fs/tools/ingest/FileSystemConverterJob.scala

```diff
@@ -113,38 +114,45 @@ trait FileSystemConverterJob extends StorageConfiguration with JobWithLibJars wi
 
       val stageCount = if (qualifiedTempPath.isDefined) { 3 } else { 2 }
 
-      var mapping = true
-      while (!job.isComplete) {
-        if (job.getStatus.getState != JobStatus.State.PREP) {
-          if (mapping) {
-            val mapProgress = job.mapProgress()
-            if (mapProgress < 1f) {
-              statusCallback(s"Map (stage 1/$stageCount): ", mapProgress, mapCounters, done = false)
-            } else {
-              statusCallback(s"Map (stage 1/$stageCount): ", mapProgress, mapCounters, done = true)
-              statusCallback.reset()
-              mapping = false
+      statusCallback match {
+        case Some(callback) =>
+          var mapping = true
+          while (!job.isComplete) {
+            if (job.getStatus.getState != JobStatus.State.PREP) {
+              if (mapping) {
+                val mapProgress = job.mapProgress()
+                if (mapProgress < 1f) {
+                  callback(s"Map (stage 1/$stageCount): ", mapProgress, mapCounters, done = false)
+                } else {
+                  callback(s"Map (stage 1/$stageCount): ", mapProgress, mapCounters, done = true)
+                  callback.reset()
+                  mapping = false
+                }
+              } else {
+                callback(s"Reduce (stage 2/$stageCount): ", job.reduceProgress(), reduceCounters, done = false)
+              }
             }
-          } else {
-            statusCallback(s"Reduce (stage 2/$stageCount): ", job.reduceProgress(), reduceCounters, done = false)
+            Thread.sleep(500)
           }
-        }
-        Thread.sleep(500)
-      }
-      statusCallback(s"Reduce (stage 2/$stageCount): ", job.reduceProgress(), reduceCounters, done = true)
+          callback(s"Reduce (stage 2/$stageCount): ", job.reduceProgress(), reduceCounters, done = true)
 
-      // Do this earlier than the data copy bc its throwing errors
-      val counterResult = (written(job), failed(job))
+          // Do this earlier than the data copy bc its throwing errors
+          val counterResult = (written(job), failed(job))
 
-      if (!job.isSuccessful) {
-        Command.user.error(s"Job failed with state ${job.getStatus.getState} due to: ${job.getStatus.getFailureInfo}")
-      } else {
-        qualifiedTempPath.foreach { tp =>
-          StorageJobUtils.distCopy(tp, metadata.getRoot, statusCallback, 3, stageCount)
-        }
-      }
+          if (!job.isSuccessful) {
+            Command.user.error(s"Job failed with state ${job.getStatus.getState} due to: ${job.getStatus.getFailureInfo}")
+          } else {
+            qualifiedTempPath.foreach { tp =>
+              StorageJobUtils.distCopy(tp, metadata.getRoot, callback, 3, stageCount)
+            }
+          }
 
-      counterResult
+          Some(counterResult)
+        case None =>
+          Command.user.info("Job Submitted")
+          if (waitForCompletion) job.waitForCompletion(false)
```

Comment:
I suspect I'm just missing the point of the ticket, but the terminology inclines me to believe that if `waitForCompletion` is true, then shouldn't `waitForCompletion(true)` be set?  In which case, can't we just use `job.waitForCompletion(waitForCompletion)`?

Reply:
I think the logic is correct - `job.waitForCompletion` will always wait before returning, the Boolean passed into it is actually a flag for 'verbose' output.

Reply:
Yeah, we should probably be better about using named params when passing in a boolean.